boxsdk/util/chunked_uploader.py

```diff
@@ -77,27 +86,42 @@ def abort(self) -> bool:
             A boolean indication success of the upload abort.
         """
         self._content_stream = None
-        self._part_array = []
-        self._inflight_part = None
+        self._part_definitions = {}
+        self._inflight_parts = {}
         self._is_aborted = True
         return self._upload_session.abort()
 
-    def _upload(self) -> None:
+    def _upload(self, futures: List[Future]) -> None:
         """
         Utility function for looping through all parts of the upload session and uploading them.
         """
-        while len(self._part_array) < self._upload_session.total_parts:
-            # Retrieve the part inflight if it exists, if it does not exist then get the next part from the stream.
-            next_part = self._inflight_part or self._get_next_part()
-            # Set the retrieve part to the current part inflight.
-            self._inflight_part = next_part
-            self._sha1.update(next_part.chunk)
-            # Retrieve the uploaded part if the part has already been uploaded. If not upload the current part.
-            uploaded_part = self._part_definitions.get(next_part.offset) or next_part.upload()
-            self._inflight_part = None
-            # Record that the part has been uploaded.
-            self._part_array.append(uploaded_part)
+        try:
+            for future in as_completed(futures):
+                future.result()
+        except Exception as exc:
+            self._executor.shutdown(wait=True)
```

Comment:
Looks like by default `wait` is set to `True` - https://docs.python.org/3/library/concurrent.futures.html
You could get rid of it if you want

Reply:
I know, but I want to make it clear that we wait